### PR TITLE
Reworks spearguns, adds Syndicate bundle, carp clothing adds carp faction

### PR DESCRIPTION
**Pull request by 81Denton**
:cl: Denton
tweak: Wearing a carp costume or hardsuit with its hood equipped now prevents users from being attacked by wild space carp.
tweak: Reworked the kinetic speargun: Its projectiles now embed and cause more damage.
add: Added a Syndicate bundle that contains a speargun, a quiver with magspears, a carp hardsuit as well as other carp accessories.
/:cl:

I reworked the speargun as a pneumatic cannon subtype - this allows spears to embed and let me control the embedding behavior. Thanks to @Tlaltecuhtli for the idea!
Magspears are now a throwing star subtype that deals increased damage.

Second change: A syndicate bundle that contains a speargun, a bandolier with 20 magspears, carp hardsuit+mask and a carp spawner grenade.

Lastly, wearing a carp costume/hardsuit with its hood adds the carp faction to players.

### CODE REVIEW

**Comment by Bawhoppen** [comment]
A bandolier is kind of a weird choice for spears. Surely some form of quiver would make more sense? Though, I'm sure you're just working with the sprites we have available.

**Comment by 81Denton** [comment]
Yeah, there aren't any quiver sprites I could have used.

**Comment by Tlaltecuhtli** [comment]
shouldnt the force depend it has a spear in it?

**Comment by 81Denton** [comment]
The actual damage is handled by the spear's throw_force and embedding behavior

**Comment by Tlaltecuhtli** [comment]
no like when you melee people with the speargun

**Comment by Shadowflame909** [comment]
1 percent chance for the grenade to spew magicarp/carp-sharks

**Comment by ExcessiveUseOfCobblestone** [comment]
@81Denton when we had bows there was a quiver sprite you can prob jerryrig

**Comment by GuyonBroadway** [comment]
Time to add a giant white space carp called "moby dickhead" or something.

**Comment by 81Denton** [comment]
@ExcessiveUseOfCobblestone good idea, I'll look into it

**Comment by davethwave** [comment]
How much damage does the speargun do?

**Comment by 81Denton** [comment]
50 damage with a guaranteed embed that deals extra damage over time.
I'm not 100% sure on the value, so it might have to be tweaked after it gets merged.

**Comment by Shadowflame909** [comment]
I think the above damage is fine but im not a maintain-it

I mean you could either get crap or the strongest non-admin spawn melee weapon in the game already, via the syndicate bundle.

**Comment by 81Denton** [comment]
Alright, it's using a quiver instead of bandolier now.

File: code/modules/clothing/spacesuits/miscellaneous.dm [hunk 1]
```diff
@@ -277,17 +277,25 @@ Contains:
...
 	helmettype = /obj/item/clothing/head/helmet/space/hardsuit/carp
 
+/obj/item/clothing/head/helmet/space/hardsuit/carp/equipped(mob/living/carbon/human/user, slot)
+	..()
+	if (slot == SLOT_HEAD)
+		user.faction |= "carp"
+
+/obj/item/clothing/head/helmet/space/hardsuit/carp/dropped(mob/living/carbon/human/user)
+	..()
+	if (user.head == src)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/clothing/spacesuits/miscellaneous.dm]: this proc is before it's removed right? If it handles removing and you call inheritance first, wouldn't this always be false?

Comment (81Denton) [reply code/modules/clothing/spacesuits/miscellaneous.dm]: That doesn't seem to be the case; when I tested it, it removed and added the "carp" faction correctly.
I copied the code for it from the prototype (RD) hardsuit:
https://github.com/tgstation/tgstation/blob/a2e4460ba583612a939cf227e4d9ebbcf1c29f82/code/modules/clothing/spacesuits/hardsuit.dm#L466-L470

Comment (ExcessiveUseOfCobblestone) [reply code/modules/clothing/spacesuits/miscellaneous.dm]: kk I didn't do my homework at the time of OP

**Pull request merged**